extensions/wikia/PhalanxII/services/PhalanxService.class.php

```diff
@@ -152,10 +152,12 @@ public function stats() {
 	 * @return integer|mixed data of blocks applied or numeric value (0 - block applied, 1 - no block applied)
 	 */
 	private function sendToPhalanxDaemon( $action, $parameters ) {
-		$baseurl = F::app()->wg->PhalanxServiceUrl;
+		global $wgConsulServiceTag, $wgConsulUrl;
+
+		$baseurl = ( new Wikia\Service\Gateway\ConsulUrlProvider( $wgConsulUrl, $wgConsulServiceTag ))->getUrl( 'phalanx' );
```

Comment:
If you take a look at ConsulUrlProvider code (getUrl method) you will notice that adresses of service instances are cached and the cache is never refreshed. Re-instantiating ConsuleUrlProvider gives as always address to instance that consul knows it is alive